### ISSUE TITLE
Extension icons sometimes not shown

### ISSUE THREAD

**Issue by eaglexiang** [microsoft/vscode-remote-release#687]
- VSCode Version:1.35.1
- Local OS Version: Linux Mint 19.1 Tessa
- Remote OS Version: CentOS Linux release 7.6.1810 (Core)
- Remote Extension/Connection Type: SSH

Steps to Reproduce:

1. open an remote folder
2. make some changes
3. plus/sub signal in git manager is not shown. where they mean staged changes or unstaged.

Does this issue occur when you try this locally?: No
Does this issue occur when you try this locally and all extensions are disabled?: No

**Comment by joaomoreno** [comment]
@eaglexiang Can you share a screenshot?

**Comment by eaglexiang** [comment]
sorry it's not must reproduced. could you wait for its next appearance?

**Comment by joaomoreno** [comment]
I'll close this for now. Just comment once more if it appears. Or open a new issue.

**Comment by eaglexiang** [comment]
![2019-06-18 17-58-26屏幕截图](https://user-images.githubusercontent.com/12654323/59739417-61339c00-9297-11e9-83b3-62f758874d14.png)

it happened again.

**Comment by eaglexiang** [comment]
as you see, `+` /  `-` icon should be shown but not.

**Comment by eaglexiang** [comment]
@joaomoreno Could you reopen this issue?

**Comment by joaomoreno** [comment]
@jrieken Not sure if missing extension icons in a remote extension are your area.

**Comment by jrieken** [comment]
> @jrieken Not sure if missing extension icons from a remote extension are your area.

👐 no clue... @eaglexiang Is this only about the +/- icons? Do other icons show? Is this something you only see when running remote?

**Comment by eaglexiang** [comment]
@jrieken yes it only happens with remote, and only misses + and –

**Comment by jrieken** [comment]
> and only misses + and –

That's interesting... Just to double check you for instance always see the commit/refresh icons even then +/- are missing?

<img width="321" alt="Screenshot 2019-06-24 at 14 54 01" src="https://user-images.githubusercontent.com/1794099/60020237-e94ae480-968f-11e9-883e-277abce0f565.png">

**Comment by eaglexiang** [comment]
![image](https://user-images.githubusercontent.com/12654323/60490029-4ec65300-9cd8-11e9-8e70-443459723209.png)


Yes you're right, they all disapeared. It happened again and I found it this time.

@jrieken Please reopen this issue by the way.

**Comment by walkafwalka** [comment]
I am getting the same issue.

Error:
```
GET vscode-remote://wsl%2Bubuntu/home/awalker/.vscode-server-insiders/bin/4f3262dd1bba349e5be1e415a60874662da2c038/extensions/git/resources/icons/dark/clean.svg net::ERR_NOT_IMPLEMENTED
```
VSCode:
```
Version: 1.37.0-insider (user setup)
Commit: 4f3262dd1bba349e5be1e415a60874662da2c038
Date: 2019-07-08T23:49:44.487Z
Electron: 4.2.5
Chrome: 69.0.3497.128
Node.js: 10.11.0
V8: 6.9.427.31-electron.0
OS: Windows_NT x64 10.0.18932
```
WSL 2:
```
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.2 LTS
Release:        18.04
Codename:       bionic
```

Posting my environment here. Hope it helps. Looks like this is a remote code issue.

**Comment by tforster** [comment]
**TL;DR; Short term solution**: Open project from `File > Open` or `File > Open Recent` and **not** from terminal with `code .` 

After a lot of trial and error it appears that the issue has to do with how VSCode is launched. If I launch from my project folder, e.g. `code .` then my Docker, AWS and Remote SSH extension icons in the activity bar are blank (e.g. space is allocated but not icon appears). Also, if I rollover filenames in the repo explorer I don't see any of the Discard Changes, Stage Changes, etc icons. With Developer Tools open and clicking on the blank spaces I get the usual ERR_NOT_IMPLEMENTED error that @walkafwalka also noted.

Now, if I close the folder with `File > Close Folder`, goto `File > Open Recent > {same project folder}` the project reopens as expected but this time all my missing icons are present. I can also confirm that straight up `File > Open` and navigating to the project folder also works.

**Comment by walkafwalka** [comment]
Awesome! Thanks @tforster. Closing the folder with `File > Close Folder` and reopening the folder via VSCode with `File > Open Recent` did the trick for me.

I also confirmed that the path for the images is still the same:
```
url("vscode-remote://wsl%2Bubuntu/home/awalker/.vscode-server-insiders/bin/46cc0a1670b7047e610aaef1bf84f1190802e9ec/extensions/git/resources/icons/dark/stage.svg")
```

So it looks the problem is that `vscode-remote://wsl%2Bubuntu/` is not referencing the correct folder on WSL machine.

**Comment by eaglexiang** [comment]
It never happened again after updating to code `v1.36.1`. I guess it owing to the local cache lately enabled.

However, I'm not sure if it will happen again in the future, and also not sure if others' issues in this issue page fixed.

**Comment by tforster** [comment]
FWIW I am running 1.36.1 and the issue persists IF I choose to open VSCode from within my project folder e.g. `code .`. The aforementioned workaround, while slightly less convenient, works.

**Comment by eamodio** [comment]
I am seeing the same thing with Remote WSL all the SCM icons are missing (and only the SCM icons). I see the behavior in both stable and the latest insiders.

**Comment by misolori** [comment]
This is also happening to extension icons in the activity bar.

**Comment by jrieken** [comment]
Plan is to migrate onto the service worker based approach that we have added for https://github.com/microsoft/vscode/issues/75061. However, that's currently blocked by https://github.com/electron/electron/issues/19150

**Comment by alexeychikk** [comment]
Running VS Code 1.36.1. @tforster Your workaround doesn't work for me.

**Comment by tforster** [comment]
@alexeychikk What are the exact steps you are following? I'm on 1.36.1 too and have been editing multiple WSL projects daily since I discovered the workaround.

**Comment by alexeychikk** [comment]
@tforster Closing folder with `File > Close Folder` helped. Thanks to @walkafwalka

**Comment by ChangeHow** [comment]
![image](https://user-images.githubusercontent.com/23733347/62837660-62d07d80-bca4-11e9-8f31-d737f82b4072.png)
problem still exist...

**Comment by kryptonian41** [comment]
The method suggested by @tforster worked for me, closing and opening the folder like he mentioned brought the icons back .

**Comment by calexandre** [comment]
Running the latest version 1.37.0 and this problem still persists. Can also confirm @tforster trick works...although its not very convenient

**Comment by matt-FFFFFF** [comment]
Also confirming this bug is still there on 1.37.0 on Windows. Close folder + Open Recent works, but is a little tiresome.

**Comment by affan-ahmed** [comment]
I hope this issue is fixed in a future release. I was thinking my extensions were not installing correctly!

**Comment by offero** [comment]
Same thing. Example here:

![image](https://user-images.githubusercontent.com/4681330/63091641-fe741f00-bf2c-11e9-8895-4491d415f341.png)

I run `vscode-insiders .` from inside of WSL 1.

**Comment by ruslanguns** [comment]
The bug only happen with Remote Connections WSL. Hopefully we get this fixed soon. 
Opening the folder manually from local instead from the WSL connection will work fine for now. Just remember that it only happens if you do "code ." from WSL not this command from Powershell / GitBash or any other which runs local windows shell.

**Comment by khpatel4991** [comment]
+1 ... on @ruslanguns approach but it would be nice to fix the core issue

**Comment by tforster** [comment]
FWIW I ave zero issue with the workaround and it's already become apart of my "muscle-memory". In fact, when it does get fixed, I am sure I will occasionally tap out the workaround out of habit. 

VSCode is such  an awesome tool and I have nothing but great things to say about it and all the people contributing to the codebase. Keep up the great work!

**Comment by ruslanguns** [comment]
Just tried with code-insiders . from wsl and still having issues.

![image](https://user-images.githubusercontent.com/19806968/63228500-bf272600-c1eb-11e9-86f2-c0cda12efcc0.png)

**Comment by alexandrudima** [comment]
@ruslanguns Thank you for trying this out with insiders. This was very recently fixed. Did you try on this build? (check the date >= 19.08)
![image](https://user-images.githubusercontent.com/5047891/63271461-7d08ed80-c29a-11e9-9410-501db9a32993.png)

Also, since nobody from our team could reproduce this issue, I'd like to ask anyone that has suffered from this in the past to please try with the latest insiders and let me know if there are still issues. 

Thank you!

**Comment by devsetgo** [comment]
I'll add my screenshot also. WSL remote.

Version: 1.37.1 (user setup)
Commit: f06011ac164ae4dc8e753a3fe7f9549844d15e35
Date: 2019-08-15T16:17:55.855Z
Electron: 4.2.7
Chrome: 69.0.3497.128
Node.js: 10.11.0
V8: 6.9.427.31-electron.0
OS: Windows_NT x64 10.0.18956

![image](https://user-images.githubusercontent.com/16019894/63291449-89d51380-c291-11e9-865d-68d17576b6d2.png)

**Comment by affan-ahmed** [comment]
@devsetgo It seems you aren't running the latest 1.38 insider version.

**Comment by devsetgo** [comment]
This is with Insider (just downloaded) and the icons are there. I have not install any plugins.

![image](https://user-images.githubusercontent.com/16019894/63292108-ee44a280-c292-11e9-8436-116d835fc335.png)

**Comment by alexandrudima** [comment]
@ricardoserradas Are you using insiders? The stable release will most likely roll out next week

If @odaibert sees missing icons in a non-remote VS Code window, then please create an issue directly against VS Code at https://github.com/microsoft/vscode and please detail exactly what VS Code version you are using, what OS, what extensions, and with a screenshot, what icons are missing.

**Comment by alexandrudima** [comment]
@r2luna The issue is fixed. The fix is available on VS Code Insiders. The fix will be available on VS Code Stable with the August 1.38 release.

**Comment by ShiftyMcCool** [comment]
I have upgraded to stable 1.38 and this issue just started for me (did not seem present in 1.37).  The trick outlined by @tforster did not work for me, the icons are always missing, although I did see them reappear at random once (unable to reproduce so far).

Windows 10 1809 / WSL
VS Code 1.38

![image](https://user-images.githubusercontent.com/1320624/64432709-130a8980-d08b-11e9-9a7f-fdb191fe1a12.png)

**Comment by alexandrudima** [comment]
@ShiftyMcCool Can you please open a new issue with the issue reporter so we can get more information about your environment? We have also changed the implementation to such an extent that it really deserves a new issue...

**Comment by ruslanguns** [comment]
In my side is fixed.
